docs/linux/dm.md

```diff
@@ -319,82 +323,87 @@ bitmaps.
 provided by the bitmaps: valid blocks are read either from the data zone or,
-if the data zone is buffered, from the buffer zone assigned to the data zone.
+if the data zone is buffered, from the buffer zone that is assigned to the data zone.
 
 ### Random Zone Reclaim
 
-After some time, the limited number of random zones available may be exhausted
-and unaligned writes to unbuffered zones become impossible. To avoid such
-situation, a reclaim process regularly scans used random zones and try to
-"reclaim" them by copying (sequentially) the valid blocks of the buffer zone
-to a free sequential zone. Once the copy completes, the chunk mapping is
-updated to point to the sequential zone and the buffer zone freed for reuse.
-
-To protect internal metadata against corruption in case of sudden power loss or
-system crash, 2 sets of metadata zones are used. One set, the primary set, is
-used as the main metadata repository, while the secondary set is used as a log.
-Modified metadata are first written to the secondary set and the log so created
-validated by writing an updated super block in the secondary set. Once this log
-operation completes, updates in place of metadata blocks can be done in the
+After some time, the limited number of available random zones may be exhausted,
+making unaligned writes to unbuffered zones impossible. To avoid this
+situation, a reclaim process regularly scans used random zones and tries to
+"reclaim" them by (sequentially) copying the valid blocks of the buffer zone to
+a free sequential zone. Once the copy completes, the chunk mapping is updated
+to point to the sequential zone and the buffer zone is freed for reuse.
+
+To protect internal metadata against corruption in cases of sudden power loss 
+or system crashes, two sets of metadata zones are used. One set (the primary 
+set) is used as the main metadata set. The other set (the secondary set) is 
+used as a log. Modified metadata are first written to the secondary set and 
+the log that is created by that act of writing to the secondary set is 
+validated by writing an updated super block in the secondary set. After this 
+log operation completes, updates in place of metadata blocks can be done in the
 primary metadata set, ensuring that one of the set is always correct.
-Flush operations are used as a commit point: upon reception of a flush
-operation, metadata activity is temporarily stopped, all dirty metadata logged
-and updated and normal operation resumed. This only temporarily delays write
-and discard requests. Read requests can be processed while metadata logging is
-executed.
+Flush operations are used as a commit point: when a flush operation is
+received, metadata activity is temporarily stopped, all dirty metadata 
+is logged and updated, and then normal operation resumes. Flush operations 
+temporarily delay write requests and discard requests. Read requests can be 
+processed while metadata logging is executed.
```

Comment:
The whole hunk is a repeat of what is in the "On-disk format" section . This should be deleted.

Reply:
I'll remove this and have it back to you tomorrow morning.

Reply:
I've removed it.

I included this here because it's a rewrite of a paragraph that appeared twice (in these places). I thought it was odd, but I also thought that it was better to present a patient with an extraneous leg rather than to amputate without checking with you first.